pkg/storage/azureblob/getter.go

```diff
@@ -0,0 +1,85 @@
+package azureblob
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/observ"
+)
+
+// Info implements the (./pkg/storage).Getter interface
+func (s *Storage) Info(ctx context.Context, module string, version string) ([]byte, error) {
+	const op errors.Op = "azureblob.Info"
+	ctx, span := observ.StartSpan(ctx, op.String())
+	defer span.End()
+
+	exists, err := s.Exists(ctx, module, version)
+	if err != nil || !exists {
+		return nil, errors.E(op, errors.M(module), errors.V(version), errors.KindNotFound)
+	}
+
+	infoReader, err := s.client.ReadBlob(ctx, config.PackageVersionedName(module, version, "info"))
+	if err != nil {
+		return nil, errors.E(op, err, errors.M(module), errors.V(version))
+	}
+	defer infoReader.Close()
```

Comment:
please don't use defer for close, we are ignoring errors returned by Close() this way.
close when done with reading and if err, then return err

Reply:
@michalpristas but what if `ioutil.ReadAll` fail? We probably will not close `infoReader` in that case. As far as I know it is safe to `defer Close` and ignore errors if we didn't write to that resource. Second solution we could leave that `defer infoReader.Close()` but also add second close `if err := infoReader.Close(); err != nil { return nil, errors.E(op, err, errors.M(module), errors.V(version))}` after `ioutil.ReadAll` as it should be safe to close it multiple times

Reply:
you're right that on non writable it is ok, but it would be good to know what happened.
if read/write fails you can call xx.Close() inside error handling block and return first error that occurred.
considering this is readable maybe we can just defer close with log message in case err happens

Reply:
I think we need to improve that case in every storage, so I would leave here as it is now and fix that for other storages at once

Reply:
i'm fine with separate PR
will create a task later